PyInstaller/depend/imphook.py

```diff
@@ -460,7 +467,10 @@ def _process_hook_func(self):
         # Call this hook() function.
         hook_api = PostGraphAPI(
             module_name=self.module_name, module_graph=self.module_graph)
-        self._hook_module.hook(hook_api)
+        try:
+            self._hook_module.hook(hook_api)
+        except ImportError:
+            raise Exception(self.exceptions.hook_import_fail())
```

Comment:
You should be raising your custom exception type here - not `Exception`.

Reply:
Again, this is not how exceptions work. Also this is a different case: "running" vs. "importing" above, so a different exception should be raised.
```suggestion
            raise ImportErrorWhenRunningHook(self.module_name, self.hook_filename)
```